### PR TITLE
Feature - AutoScale Schedules + Refactor

### PR DESCRIPTION
**Pull request by RossMurr4y**
## Description

1. Add ability to define a Schedule to an AutoScale ScaleProfile.
2. Large overhaul to ensure reusability of ScaleProfiles and move much of the logic of creating a profile into the resource functions.

## Motivation and Context

1. Schedules were missing whilst the rest of autoscale profiles were missing.
2. Multiple resources use this profile structure. Now they can all use the same Attribute structure and functions/macros with minimal overhead.

## How Has This Been Tested?

Generated and deployed local templates incl. both a bastion host with no profiles defined and a computecluster with two profiles defined.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Note on Breaking Change
Because everything else in the Scaling Profile matched the Azure resource structure and syntax, the Attribute "ScalingRules" was renamed to just "Rules" to better align with the provider.

## Followup Actions

- [x] None

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] None of the above.

### CODE REVIEW

File: azure/components/component.ftl [hunk 1]
```diff
@@ -1,5 +1,149 @@
 [#ftl]
 
+[#-- AutoScale --]
+[#assign autoScaleProfileChildrenConfiguration =
```
Comment (roleyfoley) [inline azure/components/component.ftl:4]: Is the autoscaling configuration we already have not sufficient?

File: azure/components/bastion/id.ftl [hunk 2]
```diff
@@ -2,7 +2,13 @@
...
 [@addResourceGroupInformation
   type=BASTION_COMPONENT_TYPE
-  attributes=[]
+  attributes=[
+    {
+      "Names" : "ScalingProfiles",
+      "SubObjects" : true,
+      "Children" : autoScaleProfileChildrenConfiguration
+    }
+  ]
```
Comment (roleyfoley) [inline azure/components/bastion/id.ftl:11]: Don't know if i'd put autoscaling on the bastion? Whats the user case?

File: azure/components/component.ftl (hunk 1 above)
Comment (RossMurr4y) [reply azure/components/component.ftl:4]: You're right - it should be sufficient.

What I've done so far does the job for Azure in a silo, but it not only introduces additional configuration it means that autoscaling approaches are completely different across providers. Ideally if you've worked in AWS with hamlet before and used an autoscaling component, then you should also know what to expect from the behaviour in Azure or any other provider.

I will take what I've implemented here and have it utilise the same stepped/tracked/scheduled approach that exists in the shared provider.

I'll update this PR to draft until I have that ready to go.

**Comment by RossMurr4y** [comment]
To be replaced by #194

**Pull request closed without merging**